http/src/main/java/io/micronaut/http/reactive/execution/ReactorExecutionFlowImpl.java

```diff
@@ -84,6 +88,30 @@ public ExecutionFlow<Object> putInContext(String key, Object value) {
         return this;
     }
 
+    @Override
+    public @NonNull ExecutionFlow<Object> putInContextIfAbsent(@NonNull String key, @NonNull Object value) {
+        this.value = this.value.contextWrite(context -> {
+            if (!context.hasKey(key)) {
+                return context.put(key, value);
+            } else {
+                return context;
+            }
+        });
+        return this;
+    }
+
+    @Override
+    public void cancel() {
+        List<Subscription> stc;
+        synchronized (this) {
```

Comment:
Should we avoid `synchronized`?

Reply:
synchronized is fine here because the block contains no other method calls